File: GEULGWI-BACKEND/src/main/java/com/lovedbug/geulgwi/core/domain/book/BookRepository.java
package com.lovedbug.geulgwi.core.domain.book;

import java.util.List;
import java.util.Optional;

import com.lovedbug.geulgwi.core.domain.book.entity.Book;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface BookRepository extends JpaRepository<Book, Long> {

    Optional<Book> findBookByIsbn(String isbn);

    Optional<Book> findBookByBookId(long bookId);

    boolean existsByIsbn(String isbn);

    List<Book> findTop10ByOrderByViewsDesc();

    Page<Book> findByTitleContainingIgnoreCase(String title, Pageable pageable);
}

File: GLIM-AOS/app/src/main/java/com/ssafy/glim/core/domain/repository/BookRepository.kt
package com.ssafy.glim.core.domain.repository

import com.ssafy.glim.core.domain.model.Book

interface BookRepository {

    suspend fun searchBooks(
        query: String,
        page: Int,
        searchQueryType: String
    ): List<Book>

    suspend fun updateBookViewCount(isbn: Long)

    suspend fun getBookDetail(isbn: String?, bookId: Long?): Book
}
